### PR TITLE
chores: fetch googletest via CMake

### PR DESCRIPTION
**Pull request by raphaelcoeffic**
Uses the CMake `FetchContent` module instead of `roll-your-own`.

### CODE REVIEW

File: companion/src/tests/conversions.cpp [hunk 1]
```diff
@@ -251,7 +251,7 @@ TEST(Conversions, ConversionX10From22)
   EXPECT_EQ(Board::SWITCH_3POS, settings.switchConfig[0]);
 
   EXPECT_STREQ("BT_X10", settings.bluetoothName);
-  EXPECT_STREQ("EdgeTX", settings.themeData.themeName);
+  // EXPECT_STREQ("EdgeTX", settings.themeData.themeName);
```
Comment (raphaelcoeffic) [inline companion/src/tests/conversions.cpp:254]: @elecpower not sure about these. Is that possible that the Companion unit tests were not run in GHA and thus are not really used much?

Comment (elecpower) [reply companion/src/tests/conversions.cpp:254]: The tests were run in OTX but I don't believe they are run in GHA. They had greater value when processing binary config and could pick up on data shifts.
No one has looked at them for years but that is not to say we shouldn't consider having a set of tests. With the rapid development in ETX we keep uncovering data issues.

Comment (raphaelcoeffic) [reply companion/src/tests/conversions.cpp:254]: Ok, so should I remove the `settings.themeData.*` checks, or would you want to replace that code with something more useful?

Comment (elecpower) [reply companion/src/tests/conversions.cpp:254]: Just remove

**Review by pfeerick — APPROVED** [review]

**Pull request merged**